presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -453,6 +437,41 @@ protected Node visitShowCreate(ShowCreate node, Void context)
             throw new UnsupportedOperationException("SHOW CREATE only supported for tables and views");
         }
 
+        private List<Property> buildProperties(Object objectName, StandardErrorCode errorCode,
+                Map<String, Object> properties, Map<String, PropertyMetadata<?>> allProperties)
+        {
+            if (properties.isEmpty()) {
+                return Collections.emptyList();
+            }
+
+            ImmutableSortedMap.Builder<String, Expression> sqlProperties = ImmutableSortedMap.naturalOrder();
+
+            for (Map.Entry<String, Object> propertyEntry : properties.entrySet()) {
+                String propertyName = propertyEntry.getKey();
+                Object value = propertyEntry.getValue();
+                if (value == null) {
+                    throw new PrestoException(errorCode, format("Property %s for table %s cannot have a null value", propertyName, objectName));
+                }
+
+                PropertyMetadata<?> property = allProperties.get(propertyName);
+                if (!Primitives.wrap(property.getJavaType()).isInstance(value)) {
+                    throw new PrestoException(errorCode, format(
+                            "Property %s for table %s should have value of type %s, not %s",
+                            propertyName,
+                            objectName,
```

Comment:
Using this `objectName` here is a bit hacky. It is not obvious (from reading the method body) that this method is called for column properties as well. Maybe instead of passing just `Object objectName` (`QualifiedColumnName `) you could pass an `Optional<String> columnName` together with `QualifiedObjectName objectName`.